datatypes/src/collections/feature_collection_builder.rs

```diff
@@ -253,13 +253,18 @@ where
             .builders
             .values()
             .map(|builder| {
-                let data_type_size = match builder.data_type() {
-                    arrow::datatypes::DataType::Float64 => std::mem::size_of::<f64>(),
-                    arrow::datatypes::DataType::Int64 => std::mem::size_of::<i64>(),
-                    arrow::datatypes::DataType::UInt8 => std::mem::size_of::<u8>(),
-                    arrow::datatypes::DataType::Utf8 => 0, // TODO: how to get this dynamic value
-                    _ => unreachable!("This type is not an attribute type"),
+                let data_type_size = if builder.as_any().is::<Float64Builder>() {
```

Comment:
was cleaner previously. C++ still has a `type` method on builders, ist there really no longer such u thing in rust?

Reply:
I didn't refactor it because I like the ifs more than the match block, there is just no way to get it. It would be actually even easier to get access to the underlying buffer and say `len()`, but there is no access currently.